.github/workflows/ci.yml

```diff
@@ -0,0 +1,77 @@
+name: CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+
+env:
+  IMAGE_TAG: ${{ github.sha }}
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: true
+
+      - name: Build Docker images
+        run: |
+          TARGET=prod make build
+          TARGET=dev make build
+
+      - name: Save Docker images
+        run: |
+          mkdir -p docker-image-prod docker-image-dev
+          docker save -o article-page-prod.tar libero/article-page:${IMAGE_TAG}
+          docker save -o article-page-dev.tar libero/article-page:${IMAGE_TAG}-dev
+
+      - name: Upload prod Docker image
+        uses: actions/upload-artifact@v1
+        with:
+          name: article-page-prod
+          path: article-page-prod.tar
+
+      - name: Upload dev Docker image
+        uses: actions/upload-artifact@v1
+        with:
+          name: article-page-dev
+          path: article-page-dev.tar
+
+  test:
+
+    needs: build
+    runs-on: ubuntu-latest
```

Comment:
how much should we pin this, and what are our options?

Reply:
may influence things like the Docker version, which constrains some features and/or breaks not too rarely

Reply:
I haven't been able to find a policy, but currently github only has LTS releases. So either our workflows break every two years (ubuntu-latest) or we have to bump versions in three years when 18.04 ends.

Reply:
https://github.com/libero/article-page/pull/1/commits/dc5395b60ff56f200bd8eca9dc2ca77f1fc110a6